fortanix-sgx-abi/src/lib.rs

```diff
@@ -453,6 +453,10 @@ pub const EV_RETURNQ_NOT_EMPTY: u64 = 0b0000_0000_0000_0010;
 /// An event that enclaves can use for synchronization.
 #[cfg_attr(feature = "rustc-dep-of-std", unstable(feature = "sgx_platform", issue = "56975"))]
 pub const EV_UNPARK: u64 = 0b0000_0000_0000_0100;
+/// An event that will be triggered by userspace when the cancel queue is not
+/// or no longer full.
```

Comment:
I didn't check the exact language when looking at the other queue. Are we correctly handling the “not full” case? i.e. this phrasing suggests that if the enclave calls `wait` on a queue event and there is no event pending but the queue is not empty/full, an event should be generated immediately.

Reply:
I don't think we do that in the implementation. It's based on the assumption that the enclave would only issue a `wait` usercall with `NOT_EMPTY/NOT_FULL` only after trying to send/recv on a queue. But I don't think it would be too difficult to change this. We'd need some way of getting the queue's empty/full state in `fn wait`.